openfecli/parameters/plan_network_options.py

```diff
@@ -0,0 +1,179 @@
+    class Config:
+        extra = 'allow'
+        anystr_lower = True
+
+    method: Optional[str] = None
+    settings: dict[str, Any] = {}
+
+
+class NetworkSelection(BaseModel):
+    # model_config = ConfigDict(extra='allow', str_to_lower=True)
+    class Config:
+        extra = 'allow'
+        anystr_lower = True
+
+    method: Optional[str] = None
+    settings: dict[str, Any] = {}
+
+
+class CliYaml(BaseModel):
+    # model_config = ConfigDict(extra='allow')
+    class Config:
+        extra = 'allow'
+
+    mapper: Optional[MapperSelection] = None
+    network: Optional[NetworkSelection] = None
+
+
+def parse_yaml_planner_options(contents: str) -> CliYaml:
+    """Parse and minimally validate a user provided yaml
+
+    Parameters
+    ----------
+    contents : str
+      raw yaml formatted input to parse
+
+    Returns
+    -------
+    options : CliOptions
+      will have keys for mapper and network topology choices
+
+    Raises
+    ------
+    ValueError
+      for any malformed inputs
+    """
+    raw = yaml.safe_load(contents)
+
+    if False:
+        # todo: warnings about extra fields we don't expect?
+        expected = {'mapper', 'network'}
+        for field in raw:
+            if field in expected:
+                continue
+            warnings.warn(f"Ignoring unexpected section: '{field}'")
+
+    return CliYaml(**raw)
+
+
+def load_yaml_planner_options(path: Optional[str], context) -> PlanNetworkOptions:
+    """Load cli options from yaml file path and resolve these to objects
+
+    Parameters
+    ----------
+    path : str
+      path to the yaml file
+    context
+      unused
+
+    Returns
+    -------
+    PlanNetworkOptions : namedtuple
+      a namedtuple with fields 'mapper', 'scorer', 'network_planning_algorithm',
+      and 'solvent' fields.
+      these fields each hold appropriate objects ready for use
+    """
+    from gufe import SolventComponent
+    from openfe.setup.ligand_network_planning import (
+        generate_radial_network,
+        generate_minimal_spanning_network,
+        generate_maximal_network,
+        generate_minimal_redundant_network,
+    )
+    from openfe.setup import (
+        LomapAtomMapper,
+    )
+    from openfe.setup.atom_mapping.lomap_scorers import (
+        default_lomap_score,
+    )
+    from functools import partial
+
+    if path is not None:
+        with open(path, 'r') as f:
+            raw = f.read()
+
+        # convert raw yaml to normalised pydantic model
+        opt = parse_yaml_planner_options(raw)
+    else:
+        opt = None
+
+    # convert normalised inputs to objects
+    if opt and opt.mapper:
+        mapper_choices = {
+            'lomap': LomapAtomMapper,
+            'lomapatommapper': LomapAtomMapper,
+        }
+
+        try:
+            cls = mapper_choices[opt.mapper.method]
+        except KeyError:
+            raise KeyError(f"Bad mapper choice: '{opt.mapper.method}'")
```

Comment:
It would be nice to add 
```suggestion
            raise KeyError(f"Bad mapper choice: '{opt.mapper.method}' Allowed choices are {mapper_choices}")
```

Or something like that